src/server/command/worldedit/blueprint.zig

```diff
@@ -0,0 +1,170 @@
+const std = @import("std");
+
+const main = @import("root");
+const User = main.server.User;
+const vec = main.vec;
+const Vec3i = vec.Vec3i;
+
+const openDir = main.files.openDir;
+const List = main.List;
+const Block = main.blocks.Block;
+const Blueprint = main.blueprint.Blueprint;
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+
+pub const description = "Input-output operations on blueprints.";
+pub const usage =
+	\\/blueprint save <file-name>
+	\\/blueprint delete <file-name>
+	\\/blueprint load <file-name>
+	\\/blueprint list
+;
+
+const BlueprintSubCommand = enum {
+	save,
+	delete,
+	load,
+	list,
+	other,
+	empty,
+
+	fn fromString(string: []const u8) BlueprintSubCommand {
+		return std.meta.stringToEnum(BlueprintSubCommand, string) orelse {
+			if(string.len == 0) return .empty;
+			return .other;
+		};
+	}
+};
+
+pub fn execute(args: []const u8, source: *User) void {
+	var argsList = List([]const u8).init(main.stackAllocator);
+	defer argsList.deinit();
+
+	var splitIterator = std.mem.splitScalar(u8, args, ' ');
+	while(splitIterator.next()) |a| {
+		argsList.append(a);
+	}
+
+	if(argsList.items.len < 1) {
+		source.sendMessage("#ff0000Not enough arguments for /blueprint, expected at least 1.", .{});
+		return;
+	}
+	const subcommand = BlueprintSubCommand.fromString(argsList.items[0]);
+	_ = switch(subcommand) {
+		.save => blueprintSave(argsList.items, source),
+		.delete => blueprintDelete(argsList.items, source),
+		.load => blueprintLoad(argsList.items, source),
+		.list => blueprintList(source),
+		.other => {
+			source.sendMessage("#ff0000Unrecognized subcommand for /blueprint: '{s}'", .{argsList.items[0]});
+		},
+		.empty => {
+			source.sendMessage("#ff0000Missing subcommand for **/blueprint**, usage: {s} ", .{usage});
+		},
+	} catch |err| {
+		std.log.info("Error: {s}", .{@errorName(err)});
```

Comment:
This is part of the log and should give a more specialized message.
Otherwise it will be impossible to tell where this came from.
Furthermore, should this maybe be a warning instead of an info?

Reply:
Let's just move error handling directly to source of errors.